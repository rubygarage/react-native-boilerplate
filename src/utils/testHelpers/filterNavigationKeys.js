const filterNavigationKeys = (rootRoute) => {
  if (rootRoute.routes) {
    return {
      ...rootRoute,
      routes: rootRoute.routes.map((route) => {
        const { key, ...others } = route;
        return filterNavigationKeys(others);
      }),
    };
  }
  return rootRoute;
};

export default filterNavigationKeys;
