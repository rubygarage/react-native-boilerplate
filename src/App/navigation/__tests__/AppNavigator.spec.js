import React from 'react';
import { shallow } from 'enzyme';

import filterNavigationKeys from 'utils/testHelpers/filterNavigationKeys';
import AppNavigatorWrapped from '../AppNavigator';

describe('AppNavigator component', () => {
  it('renders correctly', () => {
    const component = shallow(<AppNavigatorWrapped />);

    const { routes } = component.state().nav;
    const fixedRoutes = routes.map(filterNavigationKeys);
    const fixedNav = { ...component.state().nav, routes: fixedRoutes };
    component.setState({ nav: fixedNav });

    expect(component).toMatchSnapshot();
  });
});
