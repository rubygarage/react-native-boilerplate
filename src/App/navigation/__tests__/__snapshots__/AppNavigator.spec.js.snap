// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigator component renders correctly 1`] = `
<ContextProvider
  value="light"
>
  <ContextProvider
    value={
      Object {
        "_childrenNavigation": Object {},
        "actions": Object {},
        "addListener": [Function],
        "dangerouslyGetParent": [Function],
        "dispatch": [Function],
        "getChildNavigation": [Function],
        "getScreenProps": [Function],
        "goBack": [Function],
        "isFirstRouteInParent": [Function],
        "isFocused": [Function],
        "navigate": [Function],
        "router": Object {
          "childRouters": Object {
            "App": Object {
              "childRouters": Object {
                "MainStack": Object {
                  "childRouters": Object {
                    "Home": null,
                    "Splash screen": null,
                    "Stubs": null,
                  },
                  "getActionCreators": [Function],
                  "getActionForPathAndParams": [Function],
                  "getComponentForRouteName": [Function],
                  "getComponentForState": [Function],
                  "getPathAndParamsForState": [Function],
                  "getScreenOptions": [Function],
                  "getStateForAction": [Function],
                },
              },
              "getActionCreators": [Function],
              "getActionForPathAndParams": [Function],
              "getComponentForRouteName": [Function],
              "getComponentForState": [Function],
              "getPathAndParamsForState": [Function],
              "getScreenOptions": [Function],
              "getStateForAction": [Function],
            },
            "Initial": null,
          },
          "getActionCreators": [Function],
          "getActionForPathAndParams": [Function],
          "getComponentForRouteName": [Function],
          "getComponentForState": [Function],
          "getPathAndParamsForState": [Function],
          "getScreenOptions": [Function],
          "getStateForAction": [Function],
        },
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "routes": Array [
            Object {
              "key": "Initial",
              "params": undefined,
              "routeName": "Initial",
            },
            Object {
              "index": 0,
              "isTransitioning": false,
              "key": "App",
              "params": undefined,
              "routeName": "App",
              "routes": Array [
                Object {
                  "index": 0,
                  "isTransitioning": false,
                  "routeName": "MainStack",
                  "routes": Array [
                    Object {
                      "routeName": "Home",
                    },
                  ],
                },
              ],
            },
          ],
        },
      }
    }
  >
    <Navigator
      navigation={
        Object {
          "_childrenNavigation": Object {},
          "actions": Object {},
          "addListener": [Function],
          "dangerouslyGetParent": [Function],
          "dispatch": [Function],
          "getChildNavigation": [Function],
          "getScreenProps": [Function],
          "goBack": [Function],
          "isFirstRouteInParent": [Function],
          "isFocused": [Function],
          "navigate": [Function],
          "router": Object {
            "childRouters": Object {
              "App": Object {
                "childRouters": Object {
                  "MainStack": Object {
                    "childRouters": Object {
                      "Home": null,
                      "Splash screen": null,
                      "Stubs": null,
                    },
                    "getActionCreators": [Function],
                    "getActionForPathAndParams": [Function],
                    "getComponentForRouteName": [Function],
                    "getComponentForState": [Function],
                    "getPathAndParamsForState": [Function],
                    "getScreenOptions": [Function],
                    "getStateForAction": [Function],
                  },
                },
                "getActionCreators": [Function],
                "getActionForPathAndParams": [Function],
                "getComponentForRouteName": [Function],
                "getComponentForState": [Function],
                "getPathAndParamsForState": [Function],
                "getScreenOptions": [Function],
                "getStateForAction": [Function],
              },
              "Initial": null,
            },
            "getActionCreators": [Function],
            "getActionForPathAndParams": [Function],
            "getComponentForRouteName": [Function],
            "getComponentForState": [Function],
            "getPathAndParamsForState": [Function],
            "getScreenOptions": [Function],
            "getStateForAction": [Function],
          },
          "setParams": [Function],
          "state": Object {
            "index": 0,
            "routes": Array [
              Object {
                "key": "Initial",
                "params": undefined,
                "routeName": "Initial",
              },
              Object {
                "index": 0,
                "isTransitioning": false,
                "key": "App",
                "params": undefined,
                "routeName": "App",
                "routes": Array [
                  Object {
                    "index": 0,
                    "isTransitioning": false,
                    "routeName": "MainStack",
                    "routes": Array [
                      Object {
                        "routeName": "Home",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        }
      }
      theme="light"
    />
  </ContextProvider>
</ContextProvider>
`;
